omod/src/main/java/org/openmrs/web/controller/ForgotPasswordFormController.java

```diff
@@ -123,12 +125,20 @@ protected ModelAndView onSubmit(HttpServletRequest request, HttpServletResponse
 				finally {
 					Context.removeProxyPrivilege(PrivilegeConstants.GET_USERS);
 				}
-				
-				if (user == null /*|| StringUtils.isEmpty(user.getSecretQuestion())*/) {
+
+				String secretQuestion = null;
+				if(user != null) {
+					secretQuestion = Context.getUserService().getSecretQuestion(user);
+				}
+
+				if (user == null) {
```

Comment:
yes I will do that. Sorry you have had to repeat yourself. I was just focusing on tests